webapp/store/logic.py

```diff
@@ -378,3 +393,13 @@ def get_videos(media):
     return [
         get_video_embed_code(m["url"]) for m in media if m["type"] == "video"
     ]
+
+
+def promote_snap_with_icon(snaps):
+    snap_with_icon = next(snap for snap in snaps if snap["icon_url"] != "")
+
+    snap_with_icon_index = snaps.index(snap_with_icon)
```

Comment:
If there is no snap with icon this will raise an exception, is that intended?

Reply:
Good catch thanks